hal/targets/hal/TARGET_STM/TARGET_STM32F3/serial_api.c

```diff
@@ -69,8 +70,12 @@ static void init_uart(serial_t *obj)
     // Disable the reception overrun detection
     UartHandle.AdvancedInit.AdvFeatureInit = UART_ADVFEATURE_RXOVERRUNDISABLE_INIT;
     UartHandle.AdvancedInit.OverrunDisable = UART_ADVFEATURE_OVERRUN_DISABLE;
-
-    HAL_UART_Init(&UartHandle);
+	/* uAMR & ARM: Call to UART init is done between reset of pre-initialized variables */
```

Comment:
misaligned comment. Did you consider doing this as part of the HAL init mbed_sdk_init(), I assume this might be required for more than just UART.